src/wallet/wallet.cpp

```diff
@@ -7327,10 +7331,10 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
             }
         }
 
-        uiInterface.InitMessage(_("Rescanning..."));
+        if (!(GetBoolArg("-newwallet", false))) {uiInterface.InitMessage(_("Rescanning..."));}
         nStart = GetTimeMillis();
         walletInstance->ScanForWalletTransactions(pindexRescan, true, fRecoverMnemonic);
-        LogPrintf(" rescan      %15dms\n", GetTimeMillis() - nStart);
+        if (!(GetBoolArg("-newwallet", false))) {LogPrintf(" rescan      %15dms\n", GetTimeMillis() - nStart);}
```

Comment:
What's the reason for making the messages conditional but not the actual operation itself?

Reply:
The condition to prevent scanning if a new wallet has been created is placed inside the ScanForWalletTransactions function. This approach makes it easy to change the algorithm in the future without having to edit each function call. It also helps avoid code duplication and simplifies testing since all the logic is in one place.

Reply:
ok